artiq/applets/plot_xy.py

```diff
@@ -24,32 +31,57 @@ def data_changed(self, data, mods, title):
         fit = data.get(self.args.fit, (False, None))[1]
 
         if not len(y) or len(y) != len(x):
-            return
+            self.mismatch['X values'] = True
+            self.timer.start(1000)
+        else:
+            self.mismatch['X values'] = False
+            if not sum(self.mismatch.values()):
```

Comment:
any()

Reply:
I think `not all()` is what you mean. The timer will stop only when all the mismatches are `False`

Reply:
If use any() I think it would be `not any(list(self.mismatch.values()))` and it is not simpler than `not sum(self.mismatch.values())`

Reply:
Just `not any(self.mismatch.values())`. You don't need `list`.